### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by NovichenkoYura**

### CODE REVIEW

File: src/store/noteSlice.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+const noteSlice = createSlice({
+  name: "note",
+  initialState: {
+    notes: [],
+    value: "",
+  },
+  reducers: {
+    onAddNote(state, action) {
+      // console.log(state);
```
Comment (danya-prohvat) [inline src/store/noteSlice.js]: dont forget console.log and comments when pushing

File: src/store/noteSlice.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      // console.log(action);
+
+      state.notes.push({
+        id: uuidv4(),
+        title: "untitled note",
+        body: "",
```
Comment (danya-prohvat) [inline src/store/noteSlice.js]: better rename it in 'description'

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import { useDispatch } from "react-redux";
+import { onAddNote } from "./store/noteSlice";
+
+const App = () => {
+  const [note, setNote] = useState([]);
+  const dispatch = useDispatch();
+
+  const addNote = () => dispatch(onAddNote(note));
+
+  // const [activeNote, setActiveNote] = useState(false);
```
Comment (danya-prohvat) [inline src/App.js]: remove comments when pusing

File: src/App.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  //   const newNote = {
+  //     id: uuidv4(),
+  //     title: "untitled note",
+  //     body: "",
+  //     lastModified: Date.now(),
+  //   };
+  //   // setNotes([newNote, ...notes]);
+  // };
+
+  const onUpdateNote = (onUpdateNote) => {
```
Comment (danya-prohvat) [inline src/App.js]: delete unused code pls

File: src/Main.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  //   return <div className="no-active-note">No note selected</div>;
+
+  return (
+    <div className="app-main">
+      <div className="app-main-note-edit">
+        <input
+          type="text"
+          // id="title"
+          // value={activeNote.title}
+          onChange={(e) => {
```
Comment (danya-prohvat) [inline src/Main.js]: better take out it into separate function

File: src/Main.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            dispatch(onEditField(e.target.value));
+          }}
+          autoFocus
```
Comment (danya-prohvat) [inline src/Main.js]: why is this ?

File: src/Sidebar.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+export const Sidebar = () => {
+  //   const sortedNotes = notes.sort((a, b) => b.lastModified - a.lastModified);
+  const notes = useSelector((state) => state.notes.notes);
+  const dispatch = useDispatch();
+
+  return (
+    <div className="app-sidebar">
+      <div className="app-sidebar-header">
+        <h1>Notes</h1>
+        <button onClick={() => dispatch(onAddNote())}>Add</button>
```
Comment (danya-prohvat) [inline src/Sidebar.js]: separated function, pls

File: src/store/noteSlice.js (hunk 1 above)
Comment (NovichenkoYura) [reply src/store/noteSlice.js]: don't understand. you mean delete before pushing?

File: src/store/noteSlice.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { v4 as uuidv4 } from "uuid";
+
+const noteSlice = createSlice({
+  name: "note",
+  initialState: {
+    notes: [],
```
Comment (danya-prohvat) [inline src/store/noteSlice.js]: better notesList

File: src/store/noteSlice.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+      state.currentEditingItem = action.payload;
+    },
+
+    onReplaceEditNote(state, action) {
+      state.notes = state.notes.filter((note) => note.id !== action.payload.id);
+      state.notes.push(action.payload);
+    },
+
+    searchTitleInfo(state, action) {
+      // console.log(action.payload);
```
Comment (danya-prohvat) [inline src/store/noteSlice.js]: remove comment when push

File: src/store/noteSlice.js [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+      console.log(action.payload);
+
+      if (state.filterTitleStatus === true) {
+        state.notes = state.notes.sort((a, b) => (a.title > b.title ? 1 : -1));
+      } else
+        state.notes = state.notes.sort((a, b) => (b.title > a.title ? 1 : -1));
+      // state.filterTitleStatus !=== state.filterTitleStatus;
+    },
+
+    filterDate(state) {
```
Comment (danya-prohvat) [inline src/store/noteSlice.js]: better sortDate, because filtring is a little different

File: src/store/noteSlice.js [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+    onDeleteNote(state, action) {
+      state.notes = state.notes.filter((note) => note.id !== action.payload);
+    },
+
+    onCurrentItemInfo(state, action) {
+      state.currentEditingItem = action.payload;
+    },
+
+    onReplaceEditNote(state, action) {
```
Comment (danya-prohvat) [inline src/store/noteSlice.js:35]: you lose the order, editable contained at the end

File: src/store/noteSlice.js [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+    },
+
+    searchTitleInfo(state, action) {
+      state.searchedNotesTitle = action.payload
+     
+    },
+    
+
+    filterTitle(state) {
+      if (state.filterTitleStatus === true) {
```
Comment (danya-prohvat) [inline src/store/noteSlice.js]: just state.filterTitleStatus because filterTitleStatus already has boolean type

File: src/store/noteSlice.js [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+        state.notes = state.notes.sort((a, b) => (a.title > b.title ? 1 : -1));
+      } else
```
Comment (danya-prohvat) [inline src/store/noteSlice.js]: dont write if conditional has one line

File: src/store/noteSlice.js [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+        state.notes = state.notes.sort((a, b) => (b.title > a.title ? 1 : -1));
+      state.filterTitleStatus = !state.filterTitleStatus;
+    },
+
+    filterDate(state) {
+      if (state.filteDataStatus === true) {
+        state.notes = state.notes.sort((a, b) =>
+          a.lastModified > b.lastModified ? 1 : -1
+        );
+      } else
```
Comment (danya-prohvat) [inline src/store/noteSlice.js]: also

File: src/store/noteSlice.js [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+    filterTitle(state) {
+      if (state.filterTitleStatus === true) {
+        state.notes = state.notes.sort((a, b) => (a.title > b.title ? 1 : -1));
+      } else
+        state.notes = state.notes.sort((a, b) => (b.title > a.title ? 1 : -1));
+      state.filterTitleStatus = !state.filterTitleStatus;
+    },
+
+    filterDate(state) {
+      if (state.filteDataStatus === true) {
```
Comment (danya-prohvat) [inline src/store/noteSlice.js]: also

File: src/Sidebar.js [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+  let filterDataStatus = useSelector((state) => state.notes.filteDataStatus);
+
+  const initialNotes = useSelector((state) => state.notes.notes);
+  const searchedNotesTitle = useSelector((state) => state.notes.searchedNotesTitle);  
+  const notes = initialNotes.filter((note) => note.title.includes(searchedNotesTitle));
+
+  const dispatch = useDispatch();
+  const [activeNote, setActiveNote] = useState(false);
+
+  const addOnClick = () => {
```
Comment (danya-prohvat) [inline src/Sidebar.js]: DONT WRITE BRACKETS if one line, very unclear code

File: src/Sidebar.js [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+  filterTitle,
+  filterDate,
+} from "./store/noteSlice";
+import { useState } from "react";
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faArrowDownShortWide, faArrowUpWideShort } from '@fortawesome/free-solid-svg-icons'
+
+export const Sidebar = () => {
+  let filterTitleStatus = useSelector((state) => state.notes.filterTitleStatus);
```
Comment (danya-prohvat) [inline src/Sidebar.js]: why let??

File: src/Sidebar.js [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+  let filterDataStatus = useSelector((state) => state.notes.filteDataStatus);
```
Comment (danya-prohvat) [inline src/Sidebar.js]: don't forget about destructuring, const {filterTitleStatus, filterDataStatus} = useSelector((state) => state.notes);

File: src/Sidebar.js [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+
+  const onFilterDate = () => {
+    dispatch(filterDate());
+  };
+
+  return (
+    <div className="app-sidebar">
+      <div className="sidebar-filter">
+        <button className="sidebar-filter-btn"
+          onClick={() => {
```
Comment (danya-prohvat) [inline src/Sidebar.js]: separated onClick function

File: src/Sidebar.js [hunk 20]
```diff
@@ -0,0 +1,101 @@
...
+           
+        >
+          Title         
+          {
+            filterTitleStatus ? <FontAwesomeIcon  icon = { faArrowDownShortWide } /> : <FontAwesomeIcon  icon = { faArrowUpWideShort }/>
+         }               
+         
+        </button>
+        <button
+          onClick={() => {
```
Comment (danya-prohvat) [inline src/Sidebar.js]: also

File: src/Sidebar.js [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+        </button>
+      </div>
+
+      <div className="app-sidebar-header">
+        <h1>Notes</h1>
+        <button onClick={addOnClick}>New</button>
+      </div>
+      <div className="app-sidebar-notes">
+        {notes.map((note) => (
+          <div
```
Comment (danya-prohvat) [inline src/Sidebar.js]: destructure item element into separated component

File: src/Sidebar.js [hunk 22]
```diff
@@ -0,0 +1,101 @@
...
+            key={note.id}
+            className={`app-sidebar-note ${note.id === activeNote && "active"}`}
+            onClick={() => edititemOnClick(note.id)}
+          >
+            <div className="sidebar-note-title">
+              <strong>{note.title !== "" && note.title}</strong>
+              <button onClick={() => DeleteOnNote(note.id)}>Delete</button>
+            </div>
+            <p>{note.description !== "" && note.description}</p>
```
Comment (danya-prohvat) [inline src/Sidebar.js]: empty string has false value

File: src/Search.js [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+    initialValues: {
+      searchedtitle: "",      
+    },   
+    
+  });
+     
+  useEffect(() => {
+    if (formik.values.search) {
+      dispatch(searchTitleInfo(formik.values.search))  
+    } else {
```
Comment (danya-prohvat) [inline src/Search.js]: remove brackets

File: src/Main.js [hunk 24]
```diff
@@ -0,0 +1,112 @@
...
+import * as Yup from "yup";
+import { useMemo } from "react";
+
+import { onAddNote, onReplaceEditNote } from "./store/noteSlice";
+import { useDispatch, useSelector } from "react-redux";
+import React, { useEffect} from "react";
+
+export const Main = () => {
+  
+  const activeEditNoteId = useSelector(
```
Comment (danya-prohvat) [inline src/Main.js:10]: destructure pls, you must to have only one useSelector in component

**Pull request merged**